libs/blocks/preflight/panels/performance.js

```diff
@@ -0,0 +1,277 @@
+    key: 'personalization',
+    title: 'Personalization',
+    passed: { description: 'Personalization is currently not enabled.' },
+    failed: { description: 'MEP or Target enabled.' },
+  },
+  placeholders: {
+    key: 'placeholders',
+    title: 'Placeholders',
+    passed: { description: 'No placeholders found within the LCP section.' },
+    failed: { description: 'Placeholders found within the LCP section.' },
+  },
+  icons: {
+    key: 'icons',
+    title: 'Icons',
+    passed: { description: 'No icons found within the LCP section.' },
+    failed: { description: 'Icons found within the LCP section.' },
+  },
+};
+
+export const config = {
+  items: signal([
+    ...Object.values(text).map(({ key, title }) => ({
+      key,
+      title,
+      icon: icons.empty,
+      description: 'Loading...',
+    })),
+  ]),
+  lcp: null,
+  cls: 0,
+};
+
+export const findItem = (key) => config.items.value.find((item) => item.key === key);
+export const updateItem = ({ key, ...updates }) => {
+  const { items } = config;
+  items.value = items.value.map((item) => (item.key === key ? { ...item, ...updates } : item));
+};
+
+export function conditionalItemUpdate({ emptyWhen, failsWhen, key }) {
+  if (emptyWhen) {
+    updateItem({
+      key,
+      icon: icons.empty,
+      description: text[key].empty.description,
+    });
+    return;
+  }
+  if (failsWhen) {
+    updateItem({
+      key,
+      icon: icons.fail,
+      description: text[key].failed.description,
+    });
+    return;
+  }
+  updateItem({
+    key,
+    icon: icons.pass,
+    description: text[key].passed.description,
+  });
+}
+
+// TODO do we also want to check the content-length header?
+// https://www.w3.org/TR/largest-contentful-paint/#largest-contentful-paint-candidate-element
+// candidate’s element is a text node, or candidate’s request's response's content length
+// in bytes is >= candidate’s element's effective visual size * 0.004
+export async function checkImageSize() {
+  const { lcp } = config;
+  const hasValidImage = lcp?.url && !lcp.url.match('media_.*.mp4');
+  let blob;
+  let isSizeValid;
+  if (hasValidImage) {
+    blob = await fetch(lcp.url).then((res) => res.blob());
+    isSizeValid = blob.size / 1000 <= 100;
+  }
+
+  conditionalItemUpdate({
+    failsWhen: !isSizeValid,
+    emptyWhen: !hasValidImage,
+    key: text.imageSize.key,
+  });
+}
+
+export function checkLCP() {
+  const { lcp } = config;
+  const mainSection = document.querySelector('main > div.section');
+  const validLcp = lcp && lcp.element && lcp.url && mainSection?.contains(lcp.element);
+  conditionalItemUpdate({
+    failsWhen: !validLcp,
+    key: text.lcpEl.key,
+    icon: icons.pass,
+    description: text.lcpEl.passed.description,
+  });
+  return validLcp;
+}
+
+export const checkFragments = () => conditionalItemUpdate({
+  failsWhen: config.lcp.element.closest('.fragment') || config.lcp.element.closest('.section')?.querySelector('[data-path*="fragment"]'),
```

Comment:
Is this taking deeper nested fragments into account?

Reply:
Yes, but just in the sense of, it tells you that there is a fragment being used: https://preflight-performance-tab--milo--mokimo.hlx.page/drafts/osahin/preflight-performance/marquee-deep-nested-fragments

... we could also surface nested fragments, e.g. "amount of fragments within fragments, 5" 🤔 
Wdyt?

Reply:
I've heard of 4 levels deep nested fragments, so I think we should put this on the idea list, yes.